design-editor/src/panel/property/attribute/attribute-element.js

```diff
@@ -595,20 +595,59 @@ class Attribute extends DressElement {
      */
 	onSelectImageForCoverFlow(e) {
 		const cfFiles = e.target.files;
-		attributeUtils.copyToImagesForCoverFlow(event, cfFiles.length);
-		coverflowImage = [];
-		selectFileLen = cfFiles.length;
-		if (cfFiles) {
-			for (let i = 0; i < selectFileLen; i++) {
-				const img = cfFiles[i].name;
-				if (img.match('.jpg') || img.match('.png')) {
-					coverflowImage.push(cfFiles[i].name);
+		attributeUtils.writeMediaFileWhenIsLoaded(event, () => {
+			coverflowImage = [];
+			selectFileLen = cfFiles.length;
+			if (cfFiles) {
+				for (let i = 0; i < selectFileLen; i++) {
+					const img = cfFiles[i].name;
+					if (img.match('.jpg') || img.match('.png')) {
+						coverflowImage.push(cfFiles[i].name);
+					}
+				}
+			}
+			this.addImagesToCoverFlowWidget();
+			this._$coverflowEffect.show();
+			this._$coverflowFileSize.show();
+		});
+	}
+
+	/**
+     * Add images inside <li> tags to build coverflow widget
+     * @param {Event} e
+     */
+	addImagesToCoverFlowWidget(event) {
+		let str = '',
```

Comment:
I think these `let` variables aren't really needed in this place. 
- `str` is used only in loop and can be initialized there. 
- both `iframeElement` and `element` are used in if (selectFileLen) scope and can be initialized then. 
I think transformations like this allow make shorter and more readable functions.